src/app/datagrid/hide-show-columns/hide-show.html

```diff
@@ -0,0 +1,87 @@
+<!--
+  ~ Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+<h2>Datagrid: Hide-show columns demo</h2>
+
+<button class="btn btn-outline-primary" (click)="toggleId()">
+    Toggle ID
+</button>
+<button class="btn btn-outline-primary" (click)="toggleDate()">
+    Toggle Date
+</button>
+<clr-datagrid>
+    <clr-dg-column *ngIf="showId">
+        <ng-container *clrDgHideableColumn="{hidden: false}">
+            User ID
+        </ng-container>
+    </clr-dg-column>
+    <clr-dg-column>
+        <!--Name-->
+        <ng-container *clrDgHideableColumn="{hidden: false}">
+            Name
+        </ng-container>
+    </clr-dg-column>
+    <clr-dg-column *ngIf="showDate">
+        <ng-container *clrDgHideableColumn="{hidden: false}">
+            Creation date
+        </ng-container>
+    </clr-dg-column>
+    <clr-dg-column>
+        <ng-container *clrDgHideableColumn="{hidden: true}">
+            Pokemon
+        </ng-container>
+    </clr-dg-column>
+    <clr-dg-column>
+        <ng-container *clrDgHideableColumn>
+            Favorite color
+        </ng-container>
+    </clr-dg-column>
+
+    <clr-dg-placeholder>No users found</clr-dg-placeholder>
+
+    <clr-dg-row *clrDgItems="let user of users" [clrDgItem]="user">
+        <clr-dg-cell *ngIf="showId">{{user.id}}</clr-dg-cell>
+        <clr-dg-cell>{{user.name}}</clr-dg-cell>
+        <clr-dg-cell *ngIf="showDate">{{user.creation | date}}</clr-dg-cell>
+        <clr-dg-cell>{{user.pokemon.name}}</clr-dg-cell>
+        <clr-dg-cell>
+            <span class="color-square" [style.backgroundColor]="user.color"></span>
+        </clr-dg-cell>
+    </clr-dg-row>
+
+    <clr-dg-footer>
+        {{pagination.firstItem + 1}} - {{pagination.lastItem + 1}}
+        of {{pagination.totalItems}} users
+        <clr-dg-pagination #pagination [clrDgPageSize]="currentPageSize"></clr-dg-pagination>
+    </clr-dg-footer>
+</clr-datagrid>
+
+<h2>Datagrid: simple demo</h2>
+<!--<clr-datagrid>-->
```

Comment:
Totally on purpose aren't comments removed when we compile? har har - will remove.